### PR TITLE
Add Documentation for "User Membership" Methods

### PR DESCRIPTION
**Pull request by dannycodes**
In addition to adding documentation, undertook a minor refactor of the endpoints to better reflect the MVP use case.

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: docs/source/getting_started/projects_and_users.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+=======================
```
Comment (bfolie) [inline docs/source/getting_started/projects_and_users.rst]: Change to one sentence per line

File: docs/source/getting_started/projects_and_users.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Projects and Users
+=======================
+
+Overview
+--------
+
+Projects are containers for other resources, like datasets and modules. Access rights on resources inside a Project are
```
Comment (bfolie) [inline docs/source/getting_started/projects_and_users.rst]:
```suggestion
Projects are containers for other resources, such as datasets and modules. Access rights on resources inside a Project are
```
Previous wording could be interpreted as saying that projects are like datasets and modules in that they are all containers for other resources

File: docs/source/getting_started/projects_and_users.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+managed, granted, and revoked at the Project level.
+
+Users are literal users of the Citrine Platform. Users cannot directly access resources, but must instead be added to
```
Comment (bfolie) [inline docs/source/getting_started/projects_and_users.rst]:
```suggestion
Users are individuals using the Citrine Platform. Users cannot directly access resources, but must instead be added to
```

File: docs/source/getting_started/projects_and_users.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+Users in a Project can be listed using the :func:`~citrine.resources.project.Project.list_members` method.
+The ProjectMember array returned from this method has the user's role in the project as well as a copy
+of the User and Project objects.
+
+Add User to a Project
+--------------------
+
+Users can be added to a Project. They will be granted READ access to resources in the Project and will be given the
+role "MEMBER". This can be accomplished with the :func:`~citrine.resources.project.Project.add_user` method.
```
Comment (bfolie) [inline docs/source/getting_started/projects_and_users.rst]:
```suggestion
role "MEMBER". This is accomplished with the :func:`~citrine.resources.project.Project.add_user` method.
```

File: docs/source/getting_started/projects_and_users.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+Remove User from a Project
+-----------------------
+
+Users can also be removed from a Project. This can be achieved with the
```
Comment (bfolie) [inline docs/source/getting_started/projects_and_users.rst]:
```suggestion
Users can also be removed from a Project. This is achieved with the
```

File: src/citrine/resources/project_roles.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from typing import Union
+
+LEAD = "LEADs"
```
Comment (bfolie) [inline src/citrine/resources/project_roles.py]: Is this a typo (the "s" at the end of "LEADs")?

File: src/citrine/resources/project_member.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+class ProjectMember:
+    """A Member of a Project."""
+
+    def __init__(self,
+                 user: User,
+                 project: 'Project',
+                 role: ROLES):
+        self.user: User = user
+        self.project: 'Project' = project
```
Comment (bfolie) [inline src/citrine/resources/project_member.py]: What hath @dannycodes wrought?
You mentioned this in person, but what does this string typing do?

Comment (dannycodes) [reply src/citrine/resources/project_member.py]: https://www.python.org/dev/peps/pep-0484/#forward-references

Comment (dannycodes) [reply src/citrine/resources/project_member.py]: Lol the inline comment is clearly not where I thought it was

Comment (bfolie) [reply src/citrine/resources/project_member.py]: Wow.

File: src/citrine/resources/project_roles.py (hunk 6 above)
Comment (dannycodes) [reply src/citrine/resources/project_roles.py]: yes

File: src/citrine/resources/project_member.py (hunk 7 above)
Comment (dannycodes) [reply src/citrine/resources/project_member.py]: added a comment

**Comment by dannycodes** [comment]
@bfolie  Updated as per your comments :)

**Comment by dannycodes** [comment]
@bfolie  looks like the tests are done but haven't posted back yet. Is there a way to rerun them / prompt travis-ci to push response?

**Comment by bfolie** [comment]
> @bfolie looks like the tests are done but haven't posted back yet. Is there a way to rerun them / prompt travis-ci to push response?

Weird. I seem to remember having this problem once before and not finding a satisfactory solution (other than pushing a no-op commit)

**Comment by dannycodes** [comment]
> > @bfolie looks like the tests are done but haven't posted back yet. Is there a way to rerun them / prompt travis-ci to push response?
> 
> Weird. I seem to remember having this problem once before and not finding a satisfactory solution (other than pushing a no-op commit)

Ok I'll try that

**Pull request merged**